internal/pkg/deploy/cloudformation/stack/env.go

```diff
@@ -189,19 +189,34 @@ func (e *EnvStackConfig) telemetryConfig() *template.Telemetry {
 	}
 }
 
-func (e *EnvStackConfig) importCertARNs() []string {
+func (e *EnvStackConfig) importPublicCertARNs() []string {
 	// If a manifest is present, it is the only place we look at.
 	if e.in.Mft != nil {
 		return e.in.Mft.HTTPConfig.Public.Certificates
 	}
 	// Fallthrough to SSM config.
+	if e.in.ImportCertARNs == nil {
+		return nil
+	}
 	return e.in.ImportCertARNs
 }
 
+func (e *EnvStackConfig) importPrivateCertARNs() []string {
+	// If a manifest is present, it is the only place we look at.
+	if e.in.Mft != nil {
+		return e.in.Mft.HTTPConfig.Private.Certificates
+	}
+	// Fallthrough to SSM config.
+	if e.in.ImportVPCConfig != nil && len(e.in.ImportVPCConfig.PublicSubnetIDs) == 0 {
+		return e.in.ImportCertARNs
+	}
+	return nil
```

Comment:
```suggestion
	return e.in.ImportCertARNs
```
I _think_ we might not need to double validate this (as this is handled by env init validate).

Reply:
(see above)